runtime/src/main/java/org/corfudb/runtime/view/workflows/WorkflowRequest.java

```diff
@@ -140,9 +155,10 @@ public void invoke() {
                 else{
                     throw new IllegalStateException("Orchestrator can not be selected");
                 }
-
-            } catch (NetworkException | TimeoutException | IllegalStateException e) {
-                log.warn("WorkflowRequest: Error while running {} on attempt {}, cause:", this, x, e);
+            } catch (IllegalStateException | TimeoutException e) {
+                log.warn("WorkflowRequest: Error while running {} on attempt {}, cause:", this, retryIteration, e);
+            } catch (RuntimeException e) {
+                handleException(e);
```

Comment:
This logic here is kind of weird, why not just catch RuntimeException and check "instance of" and cause in side the catch? IllegalStateException is a RuntimeException, and TimeoutException should not be thrown without being wrapped in RuntimeException, because it's a checked exception.